galite-core/src/main/kotlin/org/kopi/galite/db/Connection.kt

```diff
@@ -87,7 +90,31 @@ class Connection {
    * Retrieves the user ID of the current user
    */
   private fun setUserID() {
-    TODO()
+
+    if (user != USERID_NO_LOOKUP) {
+      if (userName == "root" || userName == "lgvplus" || userName == "tbadmin" || userName == "dba") {
+        user = USERID_NO_LOOKUP
+      } else {
+        try {
+          val KopiUsers = object : Table("kopi_users") {
+            val id = integer("ID")
+            val kurzname = varchar("Kurzname", 255)
+          }
+
+          transaction {
+            val query = with(KopiUsers) {
+              slice(id, kurzname).select {
+                kurzname eq userName
+              }.also { if (it.empty()) throw SQLException("user unknown") }
+            }
+            user = query.also { if (it.count() > 1) throw SQLException("different users with same name") }
+                    .first()[KopiUsers.id]
+          }
+        } catch (e: SQLException) {
+          throw InconsistencyException(e.message!!)
```

Comment:
use `single()` method, and catch `NoSuchElementException` and `IllegalArgumentException` seperately with corresponding error message